helper/resource/plugin.go

```diff
@@ -77,12 +52,59 @@ func runProviderCommand(f func() error, wd *tftest.WorkingDir, opts *plugin.Serv
 		host = v
 	}
 
-	for _, ns := range namespaces {
-		reattachInfo[strings.TrimSuffix(host, "/")+"/"+
-			strings.TrimSuffix(ns, "/")+"/"+
-			providerName] = config
+	// Spin up gRPC servers for every provider factory, start a
+	// WaitGroup to listen for all of the close channels.
+	wg := sync.WaitGroup{}
+	wg.Add(len(factories))
+	reattachInfo := map[string]plugin.ReattachConfig{}
+	for providerName, factory := range factories {
+		// providerName may be returned as terraform-provider-foo, and we need
+		// just foo. So let's fix that.
+		providerName = strings.TrimPrefix(providerName, "terraform-provider-")
+
+		provider, err := factory()
+		if err != nil {
+			return fmt.Errorf("unable to create provider %q from factory: %w", providerName, err)
+		}
+
+		// PT: should this actually be called here? does it not get called by TF itself already?
+		diags := provider.Configure(ctx, terraform.NewResourceConfigRaw(nil))
```

Comment:
Have you tried it without this line? Or are you saying it was necessary to get it working

Reply:
I left it since it was already there, but I am unsure why it would be required, maybe @paddycarver knows?

Reply:
It seems like it was added in this PR? I don't think it was in the codebase before this... I can't find any reference to it.

Reply:
It came from here: https://github.com/hashicorp/terraform-plugin-sdk/blob/d609e41bce23b1dfd993591a7c3e2d5a63128dbd/helper/resource/testing.go#L505-L511

But I think lets drop it for now.